### PR TITLE
5993 dod refactor advanced search

### PR DESCRIPTION
**Pull request by codyseibert**
fixing various validation issues from UX review

Screen Shot 2020-08-10 at 5 59 37 PM

### CODE REVIEW

File: shared/src/business/entities/courtIssuedDocument/CourtIssuedDocumentTypeD.test.js [hunk 1]
```diff
@@ -33,6 +33,7 @@ describe('CourtIssuedDocumentTypeD', () => {
         freeText: 'Some free text',
         scenario: 'Type D',
       });
+      console.log('VALIDATION_ERROR_MESSAGES', VALIDATION_ERROR_MESSAGES);
```
Comment (mrinsin) [inline shared/src/business/entities/courtIssuedDocument/CourtIssuedDocumentTypeD.test.js]: I think this was left over from debugging?

**Review by mrinsin — APPROVED** [review]

File: web-client/src/presenter/actions/CaseDeadline/updateDateRangeForDeadlinesAction.js [hunk 2]
```diff
@@ -6,13 +6,27 @@ import { state } from 'cerebral';
...
-    state.screenMetadata.filterStartDateState,
-  );
-  store.set(
-    state.screenMetadata.filterEndDate,
-    state.screenMetadata.filterEndDateState,
-  );
+export const updateDateRangeForDeadlinesAction = ({
+  applicationContext,
+  get,
+  store,
```
Comment (kkoskelin) [inline web-client/src/presenter/actions/CaseDeadline/updateDateRangeForDeadlinesAction.js:12]: _nit_: jsdoc updates, probably.

File: web-client/src/views/CaseDeadlines/DateRangePickerComponent.jsx [hunk 3]
```diff
@@ -38,10 +41,50 @@ export const DateRangePickerComponent = ({
     }
   }, [dateRangePickerRef]);
 
+  useEffect(() => {
+    const startInput = document.getElementById(`${startName}-date-start`);
+    const startHiddenInput = document.querySelector(
+      `input[name="${startName}-date-start"]`,
+    );
+    if (!startValue && startInput) {
```
Comment (kkoskelin) [inline web-client/src/views/CaseDeadlines/DateRangePickerComponent.jsx:49]: Maybe we should create a devex card that submits a PR to the USWDS date-picker component, something which can call `reset()` on the component.

**Review by kkoskelin — APPROVED** [review]
OK for now, but -- would prefer to avoid `document.getElementById` and `.value = ` and `CustomEvent` in the future.  Devex task will be a good idea, I think.

**Pull request merged**